shc/web/templates/base.htm

```diff
@@ -61,18 +61,26 @@
                 <div class="ui container">
                     {% for label, icon, link in menu %}
                         {% if link is string %}
-                            <a class="mobile hidden item" href="{{ root_url }}/page/{{ link }}/">
+                            <a class="mobile hidden item {% if '/page/' ~ link ~ '/' == current_url %}selected{% endif %}"
+                               href="{{ root_url }}/page/{{ link }}/">
                                 {% if icon %}<i class="{{ icon }} icon"></i>{% endif %}
                                 {{ label }}
                             </a>
                         {% else %}
-                            <div class="mobile hidden ui dropdown item">
+                            {% set ns = namespace(isSelected=False) %}
+                            {% for sub_label, sub_icon, sub_link in link %}
+                                {% if not ns.isSelected %}
+                                    {% set ns.isSelected = (current_url == '/page/' ~ sub_link ~ '/') %}
+                                {% endif %}
+                            {% endfor %}
+                            <div class="mobile hidden ui dropdown item {% if ns.isSelected %}selected{% endif %}">
```

Comment:
Actually I would like to move all the logic from the jinja template to `interface.py` to make it easier to maintain and debug.

In my eyes the best place to temporarily store the `isSelected` information would be in the `self.ui_menu_entries`.  This could look like this:
```python
[('Label', 'icon', 'page_name','is_selected),
 ('Submenu label', None, [
    ('Label 2', 'icon', 'page_name2','is_selected'), ...
  ]),
 ...]
```
What do you think?  Any other proposal?

Reply:
Yes, I'd also prefer to evaluate the active state of the submenu entry in the Python code (that's one of the annoying things that are often required in Jinja templates but not simple to write due to the missing list comprehension / generator expressions in Jinja).

However, I would probably refactor the code a bit to use a NamedTuple or a Dataclass instead of simple tuples in the list for better readability of both, the Python code and the template code.

Reply:
Done.